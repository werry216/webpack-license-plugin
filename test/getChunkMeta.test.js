const getChunkMeta = require('../lib/getChunkMeta')
const { sep } = require('path')

describe('getChunkMeta', () => {
  test('returns names and directories of packages contained in the chunks modules', () => {
    const base = ['home', 'codepunkt', 'projects', 'monorepo'].join(sep)
    const modules = [
      [base, 'node_modules', 'package-a', 'dist'].join(sep),
      [base, 'packages', 'client'].join(sep),
      [base, 'packages', 'client', 'src'].join(sep),
      [base, 'node_modules', 'package-b'].join(sep),
      [base, 'node_modules', '@scope', 'package-c', 'dist'].join(sep),
      [base, 'node_modules', '@scope', 'package-d'].join(sep),
      [base, 'node_modules', 'package-e', 'node_modules', 'package-f'].join(sep),
    ].map(context => ({ context }))

    expect(getChunkMeta(modules)).toEqual([
      { name: 'package-a', path: [base, 'node_modules', 'package-a'].join(sep) },
      { name: 'package-b', path: [base, 'node_modules', 'package-b'].join(sep) },
      { name: '@scope/package-c', path: [base, 'node_modules', '@scope', 'package-c'].join(sep) },
      { name: '@scope/package-d', path: [base, 'node_modules', '@scope', 'package-d'].join(sep) },
      {
        name: 'package-f',
        path: [base, 'node_modules', 'package-e', 'node_modules', 'package-f'].join(sep),
      },
    ])
  })
})
