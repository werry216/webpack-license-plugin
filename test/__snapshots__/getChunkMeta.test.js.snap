// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getChunkMeta returns names and directories of packages contained in the chunks modules 1`] = `
Array [
  Object {
    "name": "package-a",
    "path": "home/codepunkt/projects/monorepo/node_modules/package-a",
  },
  Object {
    "name": "package-b",
    "path": "home/codepunkt/projects/monorepo/node_modules/package-b",
  },
  Object {
    "name": "@scope/package-c",
    "path": "home/codepunkt/projects/monorepo/node_modules/@scope/package-c",
  },
  Object {
    "name": "@scope/package-d",
    "path": "home/codepunkt/projects/monorepo/node_modules/@scope/package-d",
  },
  Object {
    "name": "package-f",
    "path": "home/codepunkt/projects/monorepo/node_modules/package-e/node_modules/package-f",
  },
]
`;
